Consistent linking 185

moekraft opened the pull request.

moekraft commented:
Updating links to internal term definitions as well as external informative and normative references. Used WAI-ARIA 1.1 as inspiration and guidance. (https://www.w3.org/TR/wai-aria-1.1/) 

- Added links to local terms once per section where they are referenced. 

- Added links to normative and informative references using bike shed keywords.

1.  On first encounter of spec, put full spec name and link externally on this name. In addition, add [normative/informative reference link] which appears at bottom of ACT document. This is generated by Bikeshed.

2. On subsequent encounters of normative/informative specs, add reference link on first instance per section.

moekraft commented:
This is to meet Issue #185 Consistent Linking of defined terms. Note: also removed italics from terms. And if a term reference is in a list, added a link.

moekraft commented:
Please review HTML version of the file, https://w3c.github.io/wcag-act/act-format-185-consistent-linking.html

skotkjerra commented:
Currently we are being inconsistent in our linking to TRs. Siteimprove propose consistently linking to non-versioned TRs. This would mean that any link to WCAG 2.0 would link to WCAG 2 e.g. https://www.w3.org/TR/WCAG/

moekraft commented:
@skotkjerra Thank you. I will work on those changes to point to use non-versioned TRs where possible. Appreciate your feedback. Moe

githubdam commented:
In section 9.1 Implementation validation, I'd recommend link text should read "test targets" instead of "pointer"

nitedog commented on act-rules-format.bs:
```diff
@@ -15,21 +15,21 @@ Markup Shorthands: markdown yes
 =====================
 
-There are currently many tools available which aid their users in testing web content for conformance to accessibility standards such as the Web Content Accessibility Guidelines (WCAG). As the web develops in both size and complexity, these tools are essential for managing the resources available on the web.
+There are currently many tools available which aid their users in testing web content for conformance to accessibility standards such as the [Web Content Accessibility Guidelines (WCAG) 2.0](https://www.w3.org/TR/WCAG20/) [[WCAG20]]. As the web develops in both size and complexity, these tools are essential for managing the resources available on the web.
```
I recommend dropping the version number "2.0", and to use this link to the overview for WCAG instead: https://www.w3.org/WAI/intro/wcag

moekraft replied:
Hi Shadi, per Stein Erik's recommendation I used the non-versioned TR link, https://www.w3.org/TR/WCAG/

moekraft replied:
As per 5/31 meeting, updating to overview page, https://www.w3.org/WAI/standards-guidelines/wcag/

nitedog commented on act-rules-format.bs:
```diff
@@ -15,21 +15,21 @@ Markup Shorthands: markdown yes
 ==============
 
-The ACT Rules Format defined in this specification is focused on the description of rules applicable to content created using web technologies, such as HTML, CSS, WAI-ARIA, SVG and more, including digital publishing. The ACT Rules Format, however, is designed to be technology agnostic, meaning that it can conceivably be used to describe test rules for other technologies.
+The ACT Rules Format defined in this specification is focused on the description of rules applicable to content created using web technologies, such as [Hypertext Markup Language](https://www.w3.org/TR/html5/) [[HTML5]], [Cascading Style Sheets](https://www.w3.org/TR/CSS2/) [[CSS21]], [Accessible Rich Internet Applications](https://www.w3.org/TR/wai-aria-1.1/) [[WAI-ARIA-1.1]], [Scaleable Vector  ...
```
Please also use this link for WAI-ARIA instead: https://www.w3.org/WAI/intro/aria

moekraft replied:
https://www.w3.org/WAI/intro/aria

nitedog commented on act-rules-format.bs:
```diff
@@ -15,21 +15,21 @@ Markup Shorthands: markdown yes
+For instance, the ACT Rules Format can be used to describe ACT Rules dedicated to testing the [accessibility requirements](#structure-accessibility-requirements) defined in WCAG, which are specifically designed for web content.
 
-Other accessibility requirements applicable to web technologies can also be testable with ACT Rules. For example, ACT Rules could be developed to test the conformance of web-based user agents to the User Agent Accessibility Guidelines. However, the ACT Rules Format would not necessarily be suitable to describe tests for the conformance of a non-web-based user agent.
+Other accessibility requirements applicable to web technologies can also be testable with ACT Rules. For example, ACT Rules could be developed to test the conformance of web-based user agents to the [User Agent Accessibility Guidelines](https://www.w3.org/TR/UAAG20/) [[UAAG20]]. However, the ACT Rules Format would not necessarily be suitable to describe tests for the conformance of a non-web-base ...
```
Also for UAAG: https://www.w3.org/WAI/intro/uaag

nitedog commented on act-rules-format.bs:
```diff
@@ -64,7 +64,7 @@ An ACT Rule MUST have a description that is in plain language and provides a bri
 -------------------------------------------------------------------
 
-An ACT Rule MUST identify the accessibility requirements to which the rule maps. For example, WCAG 2.0 Success Criterion 1.1.1. An ACT Rule is a complete or partial test for one or more accessibility requirements.
+An ACT Rule MUST identify the accessibility requirements to which the rule maps. For example, [WCAG 2.0 Success Criterion 1.1.1](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#text-equiv-all) [[WCAG20]]. An ACT Rule is a complete or partial test for one or more accessibility requirements.
```
Not certain about this, but would it be better to link to the "How to Meet WCAG 2.0" quick reference guide instead of the static TR doc? https://www.w3.org/WAI/WCAG20/quickref/#text-equiv-all

moekraft replied:
Let's raise this for discussion on Thursday's call. Thanks, Moe.

nitedog commented on act-rules-format.bs:
```diff
@@ -74,96 +74,96 @@ The actual definition of specific accessibility requirements is beyond the scope
 -------------------------------------------------------------------
 
-An ACT Rule MUST list any limitations, assumptions or any exceptions for the test, the test environment, technologies being used or the subject being tested. For example, a rule for Success Criterion 1.4.1: Use of Color has to make an assumption that CSS is used to make a link visually evident - typically by using CSS background, border, color, font, or text-decoration properties.
+An ACT Rule MUST list any limitations, assumptions or any exceptions for the test, the test environment, technologies being used or the subject being tested. For example, a rule for [WCAG 2.0 Success Criterion 1.4.1: Use of Color](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-without-color) [[WCAG20]] has to make an assumption that CSS is used to make a link visually evide ...
```
Same question here - QuickRef vs TR?

nitedog commented on act-rules-format.bs:
```diff
@@ -202,13 +202,13 @@ Rules are tested in two steps. Firstly, the applicability is used to find a list
 
 <div class=example>
-Output data using EARL and JSON-LD
+Output data using EARL and JSON-LD. (See [Evaluation and Report Language (EARL) 1.0](https://www.w3.org/TR/EARL10-Schema/) [[EARL10-Schema]] and [Java Script Object Notation (JSON)](https://www.json.org).) 
```
I suggest using this link to the overview page for EARL: https://www.w3.org/WAI/intro/earl

moekraft replied:
Done.

moekraft commented:
@skotkjerra Hi Stein Erik, I have been able to successfully update the TR links to non-versioned links, however, not all specifications have non-versioned biblio reference tags. For example, SVG comes up obsolete and CSS fails. I reverted to using SVG2 and CSS2 for the biblio references. 

Please advise. 

Thanks,
Moe

moekraft commented:
Posted updated ACT Rules Format on gh-pages, https://w3c.github.io/wcag-act/act-format-185-consistent-linking.html

Outstanding issues:
- Fixed links to go to non-versioned TR specifications, however, not all non-versioned TR specs go to the latest spec. 
-Question of when instead we should go to Working group home page
- Also for specific SC guidance, should we really go to Quick Ref?
- Lastly, not all biblio references/tags have non-versioned versions.

kasperisager commented on act-rules-format.bs:
```diff
@@ -74,96 +74,96 @@ The actual definition of specific accessibility requirements is beyond the scope
 Accessibility Support {#acc-support}
 ===========================
 
-ACT Rules are designed to test accessible uses of a web technology. However, not every part of a web technology is implemented in all assistive technologies a website may need to support. The concept of [accessibility supported](https://www.w3.org/TR/WCAG20/#accessibility-supporteddef) use of a Web technology is defined in [[WCAG20]]. Because of this, ACT Rules are not necessarily applicable in a ...
```
This paragraph still contains a reference to `WAI-ARIA-1.1`; should be `WAI-ARIA`.

moekraft replied:
Good catch. Thanks.

moekraft replied:
Waiting for confirmation from group if we should link to TR or overview page.

kasperisager commented on act-rules-format.bs:
```diff
@@ -15,21 +15,21 @@ Markup Shorthands: markdown yes
 Scope {#scope}
 ==============
 
-The ACT Rules Format defined in this specification is focused on the description of rules applicable to content created using web technologies, such as HTML, CSS, WAI-ARIA, SVG and more, including digital publishing. The ACT Rules Format, however, is designed to be technology agnostic, meaning that it can conceivably be used to describe test rules for other technologies.
```
The non-versioned SpecRef entry for SVG is simply `SVG`, which is the current W3C Recommendation.

WilcoFiers replied:
Same for CSS2, I don't think these have to link to the versioned spec.

kasperisager commented:
It might be more appropriate to link to the CSSOM spec, or the latest CSS snapshot, whenever CSS is mentioned rather than CSS2?

rdeltour commented on act-rules-format.bs:
```diff
@@ -74,96 +74,96 @@ The actual definition of specific accessibility requirements is beyond the scope
 ### HTTP Messages ### {#input-aspects-http}
 
-The HTTP messages exchanged between a client and a server as part of requesting a web page may be of interest to ACT Rules. For example, analyzing HTTP messages to perform validation of HTTP headers or unparsed HTML and CSS.
+The HTTP messages [[http11]] exchanged between a client and a server as part of requesting a web page may be of interest to ACT Rules. For example, analyzing HTTP messages to perform validation of HTTP headers or unparsed HTML [[HTML]] and CSS [[CSS2]].
```
Should we refer to HTTP 1.1 or HTTP/2?

WilcoFiers replied:
I would suggest we don't put a version in.

maryjom replied:
It would be nice not to tie to a particular version.  However, then what do we link to for the reference?

moekraft commented:
1. Updated WAI specs to point to overview pages.
2. Updated links to SC 1.1.1 and 1.4.3 to point to QuickRef
3. Updated WAI-ARIA-1.1 biblio to pull in WAI-ARIA non-versioned reference
4. Fixed broken link to test subject
